weasyl/test/searchtag/test_associate.py

```diff
@@ -0,0 +1,226 @@
+        searchtag.associate(userid_tag_adder, tags, charid=666)
+    assert err.value.value == "TargetRecordMissing"
+    
+    with pytest.raises(WeasylError) as err:
+        searchtag.associate(userid_tag_adder, tags, journalid=666)
+    assert err.value.value == "TargetRecordMissing"
+
+
+@pytest.mark.usefixtures('db')
+def test_InsufficientPermissions_WeasylError_if_user_does_not_have_tagging_permissions():
+    # Set up for this test
+    admin = db_utils.create_user()
+    userid_owner = db_utils.create_user()
+    userid_tag_adder = db_utils.create_user()
+    journalid = db_utils.create_journal(userid_owner)
+    charid = db_utils.create_character(userid_owner)
+    submitid = db_utils.create_submission(userid_owner)
+    profile.do_manage(admin, userid_tag_adder, permission_tag=False)
+    
+    with pytest.raises(WeasylError) as err:
+        searchtag.associate(userid_tag_adder, tags, submitid=submitid)
+    assert err.value.value == "InsufficientPermissions"
+    
+    with pytest.raises(WeasylError) as err:
+        searchtag.associate(userid_tag_adder, tags, charid=charid)
+    assert err.value.value == "InsufficientPermissions"
+    
+    with pytest.raises(WeasylError) as err:
+        searchtag.associate(userid_tag_adder, tags, journalid=journalid)
+    assert err.value.value == "InsufficientPermissions"
+
+
+@pytest.mark.usefixtures('db')
+def test_contentOwnerIgnoredYou_WeasylError_if_user_ignored_by_item_owner():
+    # Set up for this test
+    userid_owner = db_utils.create_user()
+    userid_tag_adder = db_utils.create_user()
+    journalid = db_utils.create_journal(userid_owner)
+    charid = db_utils.create_character(userid_owner)
+    submitid = db_utils.create_submission(userid_owner)
+    db_utils.create_ignoreuser(userid_owner, userid_tag_adder)
+    
+    with pytest.raises(WeasylError) as err:
+        searchtag.associate(userid_tag_adder, tags, submitid=submitid)
+    assert err.value.value == "contentOwnerIgnoredYou"
+    
+    with pytest.raises(WeasylError) as err:
+        searchtag.associate(userid_tag_adder, tags, charid=charid)
+    assert err.value.value == "contentOwnerIgnoredYou"
+    
+    with pytest.raises(WeasylError) as err:
+        searchtag.associate(userid_tag_adder, tags, journalid=journalid)
+    assert err.value.value == "contentOwnerIgnoredYou"
+
+
+@pytest.mark.usefixtures('db')
+def test_adding_tags_when_no_tags_previously_existed():
+    userid_owner = db_utils.create_user()
+    userid_tag_adder = db_utils.create_user()
+    journalid = db_utils.create_journal(userid_owner)
+    charid = db_utils.create_character(userid_owner)
+    submitid = db_utils.create_submission(userid_owner)
+
+    searchtag.associate(userid_tag_adder, tags, submitid=submitid)
+    searchtag.associate(userid_tag_adder, tags, charid=charid)
+    searchtag.associate(userid_tag_adder, tags, journalid=journalid)
+
+    submitid_tags = searchtag.select(submitid=submitid)
+    for tag in tags:
```

Comment:
```python
assert tags.issuperset(submitid_tags)
```

`submission_tags` might also make more sense than `submitid_tags`.

Or is this even `tags == set(submitid_tags)`?